### PR TITLE
Test and document the TLD class.

### PR DESCRIPTION
**Pull request by apmoriarty**

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    @Test
+    public void testParseFieldAndValueFromTF() {
+        Key tfKey = buildTFKey(rootId);
+        ByteSequence parsed = TLD.parseFieldAndValueFromTF(tfKey.getColumnQualifierData());
+        ByteSequence expected = new ArrayByteSequence(field + '\u0000' + value);
+        assertEquals(expected, parsed);
+        
+        tfKey = buildTFKey(childId);
+        parsed = TLD.parseFieldAndValueFromTF(tfKey.getColumnQualifierData());
+        expected = new ArrayByteSequence(field + '\u0000' + value);
```
Comment (jschmidt10) [inline warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java]: This `expected` is the same every time.

Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java]: Updated!

File: warehouse/query-core/src/main/java/datawave/query/tld/TLD.java [hunk 2]
```diff
@@ -43,11 +92,18 @@ public static ByteSequence parseFieldAndValueFromFI(ByteSequence cf, ByteSequenc
...
+     * @return - a byte sequence that is the field and value separated by a null byte
+     */
     public static ByteSequence parseFieldAndValueFromTF(ByteSequence cq) {
-        ArrayList<Integer> deezNulls = instancesOf(0, cq, 2);
-        ArrayList<Integer> deezOtherNulls = lastInstancesOf(0, cq, 1);
-        final int startFn = deezOtherNulls.get(0) + 1, stopFn = cq.length();
-        final int startFv = deezNulls.get(1) + 1, stopFv = deezOtherNulls.get(0);
+        ArrayList<Integer> nulls = instancesOf(0, cq, 2);
+        ArrayList<Integer> otherNulls = lastInstancesOf(0, cq, 1);
+        final int startFn = otherNulls.get(0) + 1, stopFn = cq.length();
```
Comment (jschmidt10) [inline warehouse/query-core/src/main/java/datawave/query/tld/TLD.java]: Not your typo, but I feel like this was intended to be `startFv` for "field/value"

Comment (jschmidt10) [reply warehouse/query-core/src/main/java/datawave/query/tld/TLD.java]: Ah, should've read the next line. Nevermind.

File: warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        Key fiKey = buildFiKey(rootId);
+        Text shard = fiKey.getRow();
+        ByteSequence id = new ArrayByteSequence(rootId);
+        ByteSequence cq = fiKey.getColumnQualifierData();
+        Text cv = fiKey.getColumnVisibility();
+        long ts = fiKey.getTimestamp();
+        
+        Key parentKey = TLD.buildParentKey(shard, id, cq, cv, ts);
+        String cq2 = value + '\u0000' + datatype + '\u0000' + rootId;
+        Key expectedKey = new Key(row, rootId, cq2, fiKey.getTimestamp());
```
Comment (jschmidt10) [inline warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java]: You could use your `ts` variable for these `fiKey.getTimestamp()` calls.

Comment (ivakegg) [reply warehouse/query-core/src/test/java/datawave/query/tld/TLDTest.java]: +1

**Pull request merged**